src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -77,6 +78,22 @@
         type="function_name_identifier",
     ),
     DatatypeIdentifierSegment=Ref("SingleIdentifierGrammar"),
+    PrimaryKeyGrammar=Sequence(
+        "PRIMARY", "KEY", OneOf("CLUSTERED", "NONCLUSTERED", optional=True)
+    ),
+    FromClauseTerminatorGrammar=OneOf(
+        "WHERE",
+        "LIMIT",
+        "GROUP",
+        "ORDER",
+        "HAVING",
+        "QUALIFY",
```

Comment:
Do we need QUALIFY and WINDOW here?  Pretty sure they are not recognized in TSQL.

Reply:
No we dont need them :)